02_commands.html

```diff
@@ -1,206 +1,524 @@
+
+      <li><p>Let <var>request match</var> be the result of <a>matching a request</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> and <a href="https://fetch.spec.whatwg.org/#concept-request-url" class="externalRef">url</a> as arguments.</p></li>
+
+      <li><p>If <var>request match</var> is of type <a>error</a>, <a>return an error</a> with <var>request match</var>'s <a>error code</a> and jump to step 1.</p>
+
+        <p>Otherwise, let <var>command</var>, <var>session id</var> and <var>element id</var> be <var>request match</var>'s data.</p></li>
+
+      <li><p>If <var>command</var> is not <a>new session</a>:</p>
+        <ol>
+          <li><p>If <var>session id</var> is not equal to the <a>id</a> of any session in the <a>list of active sessions</a>, <a>return an error</a> with <a>error code</a> <a>invalid session id</a>, then jump to step 1.</p>
+
+          <p>Otherwise, let the <a>current session</a> be the session with <a>id</a> <var>session id</var></p>
+          </li>
+        </ol>
+      </li>
+
+      <li><p>If <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> is POST:</p>
+        <ol>
+
+          <li><p>Let <var>success</var> and <var>parameters</var> be the result of <a>parsing as JSON</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-body" class="externalRef">body</a> as the argument.</p></li>
+
+          <li><p>If <var>success</var> is false, or <var>success</var> is true and <var>parameters</var> is not an Object object, <a>return an error</a> with status <a>error code</a> <a>invalid argument</a> and jump back to step 1 in this overall algorithm.</p></li>
+        </ol>
+
+        <p>Otherwise, let <var>parameters</var> be null</p>
+      </li>
+
+      <li><p>Let <var>response data</var> be a <a>command response</a> object obtained by running running the steps defined by <var>command</var> with arguments <var>element id</var> and <var>parameters</var>.</p></li>
+
+      <li><p>If <var>response data</var> is an <a>error</a>, <a>return an error</a> with <a>error code</a> equal to <var>response data</var>'s <a>error code<a>.</p>
+
+        <p>Otherwise <a>send a response</a> with status 200 and <var>response data</var>'s data.</p></li>
+
+      <li><p>Jump to step 1</p></li>
+    </ol>
+
+    <p>When required to <dfn>send a response</dfn>, with arguments <var>status</var> and <var>data</var>, a remote end must run the following steps:</p>
+
+    <ol>
+      <li><p>Let <var>response</var> be a new <a href="https://fetch.spec.whatwg.org/#concept-response" class="externalRef">response</a>.</p></li>
```

Comment:
I'm curious here whether we have to say anything about mutability or not.  I see other specs using prose like “initially left empty” when talking about objects that are created but which can be modified later.